File: backend/pkg/db/models/appointment.go
package models

import "time"

type Appointment struct {
	ID          uint      `json:"id,omitempty" gorm:"primary_key"`
	DoctorID    uint      `json:"doctor_id" validate:"required"`
	Doctor      Doctor    `json:"doctor" validate:"-"`
	UserID      uint      `json:"user_id"`
	User        User      `json:"user" validate:"-"`
	StartTime   time.Time `json:"start_time" validate:"required"`
	EndTime     time.Time `json:"end_time" validate:"required"`
	Duration    string    `json:"duration" validate:"required"`
	Status      string    `json:"status" gorm:"default:scheduled"`
	MeetingLink string    `json:"meeting_link"`
	IsDeleted   bool      `json:"is_deleted" gorm:"default:false"`
	CreatedAt   time.Time `json:"created_at,omitempty" gorm:"autoCreateTime"`
	UpdatedAt   time.Time `json:"updated_at,omitempty" gorm:"autoUpdateTime"`
}

type RescheduleAppointmentType struct {
	StartTime time.Time `json:"start_time" validate:"required"`
	EndTime   time.Time `json:"end_time" validate:"required"`
}

File: frontend/app/types/appointment.ts
import { IUser } from '.'
import { IDoctor } from './health-provider'

export interface IAppointment {
  date: string
  appointments: [
    {
      id: number
      doctor_id: number
      doctor: IDoctor
      user_id: number
      user: IUser
      start_time: string
      end_time: string
      duration: string
      status: string
      meeting_link: string
      is_deleted: boolean
      created_at: string
      updated_at: string
    },
  ]
}
